Exiled.API/Features/Room.cs

```diff
@@ -37,8 +38,16 @@ public Room(string name, Transform transform, Vector3 position)
             Type = FindType(name);
             Doors = FindDoors();
             flickerableLightController = transform.GetComponentInChildren<FlickerableLightController>();
+
+            Id = IdCounter;
+            IdCounter++;
         }
 
+        /// <summary>
+        /// Gets the room's index in <see cref="Map.Rooms"/>.
```

Comment:
I don't think this is confusing at all. The only time that new Room call happens is when Map.Rooms doesn't contain  the room they are looking for, it will return a new room with a unique Id. Though that -Should never happen- if Map.Rooms work properly. The only change we could make here is adding the new room to Map.Rooms.

Reply:
Don't think that our rooms API is done. It doesn't work as expected and in most cases returns a new room.

Reply:
hmm yeah you're right. I'll implement a fix for that in another branch, sec.